crates/lib3h/src/engine/real_engine.rs

```diff
@@ -544,7 +544,7 @@ impl<T: Transport, D: Dht> RealEngine<T, D> {
         output.push(Lib3hServerProtocol::SuccessResult(res));
         // First create DhtConfig for space gateway
         let agent_id: String = join_msg.agent_id.clone().into();
-        let this_net_peer = self.network_gateway.borrow().this_peer().clone();
+        let this_net_peer = self.network_gateway.as_ref().this_peer().clone();
         let this_peer_transport_id_as_uri =
             // TODO #175 - encapsulate this conversion logic
             Url::parse(format!("transportId:{}", this_net_peer.peer_address.clone()).as_str()).unwrap();
```

Comment:
What do you think about something like that instead of those two clone calls : 
```rust
        let this_peer_transport_id_as_uri = {
            let this_net_peer_address = self.network_gateway.as_ref();
            Url::parse(format!("transportId:{}", this_net_peer_address.this_peer().peer_address).as_str())
                .expect("Fail to parse peer address.")
        };
```

Reply:
nice :+1: - updated